crates/astria-sequencer-relayer/src/finalization_pipeline/head_candidate.rs

```diff
@@ -0,0 +1,81 @@
+use astria_sequencer_types::SequencerBlockData;
+use tendermint::Hash;
+use tendermint_rpc::endpoint::block;
+
+#[derive(Clone, Default, Debug)]
+pub(crate) enum HeadCandidate {
+    ProposedByValidator(SequencerBlockData),
+    CometBft(CometBftHeadCandidate),
+    #[default]
+    Default,
+}
+
+impl TryInto<SequencerBlockData> for HeadCandidate {
+    type Error = ();
+
+    fn try_into(self) -> Result<SequencerBlockData, Self::Error> {
+        use HeadCandidate::*;
+        match self {
+            ProposedByValidator(block) => Ok(block),
+            CometBft(_) => Err(()),
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl HeadCandidate {
+    pub(crate) fn new_from_validator(block: SequencerBlockData) -> Self {
+        Self::ProposedByValidator(block)
+    }
+
+    pub(crate) fn new_from_cometbft(block: block::Response) -> Self {
+        Self::CometBft(block.into())
+    }
+
+    pub(crate) fn block_hash(&self) -> Hash {
+        use HeadCandidate::*;
```

Comment:
is this needed?

Reply:
not necessarily, thought it made the code in the finalisation pipeline easier to read. are you strongly opposed to it instead of calling `block.block.block_hash()` ?

Reply:
oh yeah this is needed! matching on the enum variant to get the block hash in the pipeline code is to granular to have a good overview of the pipeline code